pinot-query-planner/src/main/java/org/apache/pinot/query/planner/physical/DispatchablePlanMetadata.java

```diff
@@ -35,18 +35,28 @@
 /**
  * The {@code DispatchablePlanMetadata} info contains the information for dispatching a particular plan fragment.
  *
- * <p>It contains information aboute:
+ * <p>It contains information
  * <ul>
- *   <li>the tables it is suppose to scan for</li>
- *   <li>the underlying segments a stage requires to execute upon.</li>
- *   <li>the server instances to which this stage should be execute on</li>
+ *   <li>extracted from {@link org.apache.pinot.query.planner.physical.DispatchablePlanVisitor}</li>
+ *   <li>extracted from {@link org.apache.pinot.query.planner.physical.PinotDispatchPlanner}</li>
  * </ul>
  */
 public class DispatchablePlanMetadata implements Serializable {
-  // These 2 fields are extracted from TableScanNode
+
+  // --------------------------------------------------------------------------
+  // Fields extracted with {@link DispatchablePlanVisitor}
+  // --------------------------------------------------------------------------
+  // info from TableNode
   private final List<String> _scannedTables;
   private Map<String, String> _tableOptions;
+  // info from MailboxSendNode - whether a stage is pre-partitioned by the same way the sending exchange desires
+  private boolean _isPrePartitioned;
```

Comment:
(minor) I prefer `_isPartitioned` over `_isPrePartitioned`. What does `pre` stand for?

Reply:
"pre" stands for the data input to the mailbox is "already" partitioned --> isPartition is confusing b/c it can also mean the result of the mailbox is partitioned

Reply:
`prePartitioned` in mailbox makes sense, but is strange in stage metadata. Based on my understanding, it stands for the data within a stage being partitioned

Reply:
yeah `prePartitioned` is not the same as `partitioned`. it means data in this stage is `partitioned` and the root of this stage (sender) is sending data with the same partitioning.

Reply:
I see. I missed the javadoc that this is just storing the property of `MailboxSendNode`